skyportal/handlers/api/mpc.py

```diff
@@ -0,0 +1,197 @@
+        except Exception:
+            return self.error('Cannot read in search radius.')
+
+        obscode = data.get('obscode', '500')
+
+        with self.Session() as session:
+            obj = session.scalars(
+                Obj.select(session.user_or_token, mode='update').where(Obj.id == obj_id)
+            ).first()
+            if obj is None:
+                return self.error(f"Cannot find object with ID {obj_id}.")
+
+            year = date.strftime("%Y")
+            month = date.strftime("%m")
+            hour = int(date.strftime("%H"))
+            minute = int(date.strftime("%M"))
+            second = float(date.strftime("%S.%f"))
+            day = f"{int(date.strftime('%d'))+(hour+minute/60+second/3600)/24:.2f}"
+
+            ra = Angle(obj.ra, unit='degree')
+            ra_hms = ra.to_string(
+                unit='hourangle', sep=' ', precision=2, pad=True
+            ).split(" ")
+            dec = Angle(obj.dec, unit='degree')
+            dec_dms = dec.to_string(unit=u.deg, sep=' ', precision=1, pad=True).split(
+                " "
+            )
+
+            url = f"{mpcheck_url}?year={year}&month={month}&day={day}&which=pos&ra={ra_hms[0]}+{ra_hms[1]}+{ra_hms[2]}&decl={dec_dms[0]}+{dec_dms[1]}+{dec_dms[2]}&TextArea=&radius={search_radius}&limit={limiting_magnitude}&oc={obscode}&sort=d&mot=h&tmot=s&pdes=u&needed=f&ps=n&type=p"
+
+            try:
+                loop = asyncio.get_event_loop()
+            except Exception:
+                loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(loop)
+            IOLoop.current().run_in_executor(
+                None,
+                lambda: query_mpc(obj_id, self.associated_user_object.id, url),
+            )
+
+            return self.success()
+
+
+def query_mpc(obj_id, user_id, url):
+    """Query MPC for a given object.
+    obj_id : str
+        Object ID
+    user_id : int
+        SkyPortal ID of User posting the MPC result
+    url : str
+        MPC query URL
+    """
+
+    if Session.registry.has():
+        session = Session()
+    else:
+        session = Session(bind=DBSession.session_factory.kw["bind"])
+
+    log(f'Querying MPC for {obj_id}: {url}')
+
+    try:
+        user = session.query(User).get(user_id)
+
+        obj = session.scalars(Obj.select(user).where(Obj.id == obj_id)).first()
+
+        requests_session = requests.Session()
+        response = requests_session.get(url)
+        soup = BeautifulSoup(response.text, "html.parser")
+
+        pre = soup.find("pre")
+        if pre is None:
+            log(f'Error checking MPC for {obj_id}: {str(soup)}')
+        else:
+            if re.findall("The following objects,.*", response.text)[0]:
+                responseSplit = response.text.split("(1)")[-1].split(" ")
+                mpc_name = list(filter(None, responseSplit))[0]
+
+                log(f'{obj_id}: identified MPC name {mpc_name}')
+
+                obj.is_roid = True
```

Comment:
What does `roid` stand for? is that short for asteroid...?

Reply:
Yeah... that was historical...